pkg/controllers/machine/termination/suite_test.go

```diff
@@ -0,0 +1,183 @@
+	"github.com/aws/karpenter-core/pkg/controllers/machine/termination"
+	"github.com/aws/karpenter-core/pkg/operator/controller"
+	"github.com/aws/karpenter-core/pkg/operator/scheme"
+	. "github.com/aws/karpenter-core/pkg/test/expectations"
+
+	"github.com/aws/karpenter-core/pkg/test"
+)
+
+var ctx context.Context
+var env *test.Environment
+var fakeClock *clock.FakeClock
+var cloudProvider *fake.CloudProvider
+var machineController controller.Controller
+var terminationController controller.Controller
+
+func TestAPIs(t *testing.T) {
+	ctx = TestContextWithLogger(t)
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Machine")
+}
+
+var _ = BeforeSuite(func() {
+	fakeClock = clock.NewFakeClock(time.Now())
+	env = test.NewEnvironment(scheme.Scheme, test.WithCRDs(apis.CRDs...), test.WithFieldIndexers(func(c cache.Cache) error {
+		return c.IndexField(ctx, &v1.Node{}, "spec.providerID", func(obj client.Object) []string {
+			return []string{obj.(*v1.Node).Spec.ProviderID}
+		})
+	}))
+	ctx = settings.ToContext(ctx, test.Settings())
+	cloudProvider = fake.NewCloudProvider()
+	machineController = machine.NewController(fakeClock, env.Client, cloudProvider)
+	terminationController = termination.NewController(env.Client, cloudProvider)
+})
+
+var _ = AfterSuite(func() {
+	Expect(env.Stop()).To(Succeed(), "Failed to stop environment")
+})
+
+var _ = AfterEach(func() {
+	fakeClock.SetTime(time.Now())
+	ExpectCleanedUp(ctx, env.Client)
+	cloudProvider.Reset()
+})
+
+var _ = Describe("Termination", func() {
+	var provisioner *v1alpha5.Provisioner
+	var machine *v1alpha5.Machine
+
+	BeforeEach(func() {
+		provisioner = test.Provisioner()
+		machine = test.Machine(v1alpha5.Machine{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					v1alpha5.ProvisionerNameLabelKey: provisioner.Name,
+				},
+			},
+			Spec: v1alpha5.MachineSpec{
+				Resources: v1alpha5.ResourceRequirements{
+					Requests: v1.ResourceList{
+						v1.ResourceCPU:          resource.MustParse("2"),
+						v1.ResourceMemory:       resource.MustParse("50Mi"),
+						v1.ResourcePods:         resource.MustParse("5"),
+						fake.ResourceGPUVendorA: resource.MustParse("1"),
+					},
+				},
+			},
+		})
+	})
+	It("should delete the node and the CloudProvider Machine when Machine deletion is triggered", func() {
+		ExpectApplied(ctx, env.Client, provisioner, machine)
+		ExpectReconcileSucceeded(ctx, machineController, client.ObjectKeyFromObject(machine))
+
+		machine = ExpectExists(ctx, env.Client, machine)
+		_, err := cloudProvider.Get(ctx, machine.Status.ProviderID)
+		Expect(err).ToNot(HaveOccurred())
+
+		node := test.MachineLinkedNode(machine)
+		ExpectApplied(ctx, env.Client, node)
+
+		// Expect the node and the machine to both be gone
+		Expect(env.Client.Delete(ctx, machine)).To(Succeed())
+		ExpectReconcileSucceeded(ctx, terminationController, client.ObjectKeyFromObject(machine)) // triggers the node deletion
+		ExpectFinalizersRemoved(ctx, env.Client, node)
```

Comment:
I think we should assert that the nodes are still found before removing finalizers, right?
```suggestion
                ExpectExists(ctx, env.Client, node)
		ExpectFinalizersRemoved(ctx, env.Client, node)
```

Reply:
That seems to just be validating that finalizers just work, which feels outside the scope of this PR

Reply:
To me, it validates that the finalizers exist, which is part of the termination controller logic, right?